core/src/tasks/settle_match_internal.rs

```diff
@@ -253,22 +292,255 @@ impl SettleMatchInternalTask {
+                valid_settle_proof.statement.party1_modified_shares.clone(),
+                self.order1_proof.clone(),
+                self.order2_proof.clone(),
+                valid_match_proof,
+                valid_settle_proof,
+            )
+            .await
+            .map_err(|err| SettleMatchInternalTaskError::Starknet(err.to_string()))?;
+
+        let tx_info = self
+            .starknet_client
+            .poll_transaction_completed(tx_hash)
+            .await
+            .map_err(|err| SettleMatchInternalTaskError::Starknet(err.to_string()))?;
+
+        // Check transaction status
+        if let TransactionStatus::Rejected = tx_info.status {
+            return Err(SettleMatchInternalTaskError::Starknet(format!(
+                "transaction rejected: {:?}",
+                tx_info
+                    .transaction_failure_reason
+                    .unwrap_or(TransactionFailureReason {
+                        code: "".to_string(),
+                        error_message: Some(ERR_UNKNOWN_TX_FAILURE.to_string())
+                    })
+            )));
+        }
+
+        Ok(())
     }
 
     /// Update the wallet state and Merkle openings
     async fn update_state(&self) -> Result<(), SettleMatchInternalTaskError> {
-        todo!()
+        // Lookup the wallets that manage each order
+        let wallet1 = self
+            .find_wallet_for_order(&self.order_id1)
+            .await
+            .ok_or_else(|| {
+                SettleMatchInternalTaskError::MissingState(ERR_WALLET_NOT_FOUND.to_string())
+            })?;
+        let wallet2 = self
+            .find_wallet_for_order(&self.order_id2)
+            .await
+            .ok_or_else(|| {
+                SettleMatchInternalTaskError::MissingState(ERR_WALLET_NOT_FOUND.to_string())
+            })?;
+
+        // Update the wallet state
+        let (mut buy_side_wallet, buy_side_order, mut sell_side_wallet, sell_side_order) =
+            match scalar_to_u64(&self.match_result.direction.val) {
+                0 => (wallet1, self.order_id1, wallet2, self.order_id2),
+                1 => (wallet2, self.order_id2, wallet1, self.order_id1),
+                _ => panic!("invalid match direction"),
+            };
+
+        // Update the balances
+        let base_mint = scalar_to_biguint(&self.match_result.base_mint.val);
+        let quote_mint = scalar_to_biguint(&self.match_result.quote_mint.val);
+
+        let base_amount = scalar_to_u64(&self.match_result.base_amount.val) as i64;
+        let quote_amount = scalar_to_u64(&self.match_result.quote_amount.val) as i64;
+
+        Self::update_balance(base_mint.clone(), base_amount, &mut buy_side_wallet);
+        Self::update_balance(base_mint, -base_amount, &mut sell_side_wallet);
+        Self::update_balance(quote_mint.clone(), -quote_amount, &mut buy_side_wallet);
+        Self::update_balance(quote_mint, -quote_amount, &mut sell_side_wallet);
```

Comment:
```suggestion
        Self::update_balance(quote_mint, quote_amount, &mut sell_side_wallet);
```
sell side buys the quote, right?